integrations/access/accessrequest/app.go

```diff
@@ -488,6 +496,20 @@ func (a *App) updateMessages(ctx context.Context, reqID string, tag pd.Resolutio
 	return nil
 }
 
+func (a *App) getLoginsByRole(ctx context.Context, req types.AccessRequest) (map[string][]string, error) {
+	loginsByRole := make(map[string][]string, len(req.GetRoles()))
+
+	for _, role := range req.GetRoles() {
+		currentRole, err := a.apiClient.GetRole(ctx, role)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		loginsByRole[role] = currentRole.GetLogins(types.Allow)
```

Comment:
Couple of questions here:

- If a role has a login template variable (e.g. `{{external.logins}}`) would we should it as-is?
- Role can have denied logins too, should we display them as well?

Reply:
The template variable will be displayed as is. We do not filter denied logins. I can look into improving these two cases in a follow up PR.